File: Golang/Medium/2125_Number_of_Laser_Beams_in_a_Bank/solution.go
package numberoflaserbeamsinabank

func numberOfBeams(bank []string) int {
	bankArray := []int{}

	for r, _ := range bank {
		count := 0
		for c, _ := range bank[0] {
			if bank[r][c] == '1' {
				count += 1
			}
		}
		if count != 0 {
			bankArray = append(bankArray, count)
		}
	}

	total := 0
	for i := 0; i < len(bankArray)-1; i++ {
		total += bankArray[i] * bankArray[i+1]
	}
	return total
}

File: Python/Medium/2125_Number_of_Laser_Beams_in_a_Bank/solution.py
class Solution:
    def numberOfBeams(self, bank: list[str]) -> int:
        bankArray = []

        for r in range(len(bank)):
            count = 0
            for c in range(len(bank[0])):
                if bank[r][c] == "1":
                    count += 1
            if count != 0:
                bankArray.append(count)

        total = 0
        for i in range(len(bankArray)-1):
            total += bankArray[i] * bankArray[i+1]
        return total